pkg/storage/mvcc_value.go

```diff
@@ -239,6 +239,35 @@ func EncodeMVCCValueToBuf(v MVCCValue, buf []byte) ([]byte, bool, error) {
 	return buf, true, nil
 }
 
+func mvccValueSize(v MVCCValue) (size int, extendedEncoding bool) {
+	if v.MVCCValueHeader.IsEmpty() && !disableSimpleValueEncoding {
+		return len(v.Value.RawBytes), false
+	}
+	return extendedPreludeSize + v.MVCCValueHeader.Size() + len(v.Value.RawBytes), true
+}
+
+func encodeExtendedMVCCValueToSizedBuf(v MVCCValue, encodedValueLen int, buf []byte) error {
+	headerSize := encodedValueLen - len(v.Value.RawBytes)
+	headerLen := headerSize - extendedPreludeSize
+	// Extended encoding. Wrap the roachpb.Value encoding with a header containing
+	// MVCC-level metadata. Requires a copy.
+	// 4-byte-header-len
+	binary.BigEndian.PutUint32(buf, uint32(headerLen))
+	// 1-byte-sentinel
+	buf[tagPos] = extendedEncodingSentinel
+	// mvcc-header
+	//
+	// NOTE: we don't use protoutil to avoid passing v.MVCCValueHeader through
+	// an interface, which would cause a heap allocation and incur the cost of
+	// dynamic dispatch.
```

Comment:
I could see having a function level comment just telling the reader to refer to the comments on EncodeMVCCValueToBuf for details info.